src/git-version.cr

```diff
@@ -0,0 +1,141 @@
+require "file_utils"
+
+require "semantic_version"
+
+module GitVersion
+  extend self
+
+  BASE_VERISON_STRING = "0.0.0"
+  BASE_VERISON        = SemanticVersion.parse(BASE_VERISON_STRING)
+
+  DEV_BRANCH_SUFFIX = "SNAPSHOT"
+
+  MASTER_BRANCH = "master"
+
+  MAJOR_BUMP_COMMENT = "breaking:"
+  MINOR_BUMP_COMMENT = "feature:"
+
+  class Git
+    def initialize(@devBranch : String, @folder = FileUtils.pwd)
+      #
+    end
+
+    private def exec(cmd)
+      strout = IO::Memory.new
+
+      if !Process.run(
+           command: cmd,
+           shell: true,
+           output: strout,
+           error: Process::Redirect::Close,
+           chdir: @folder
+         ).success?
+        raise "[ERROR] Command #{cmd} failed."
+      end
+
+      return strout.to_s.split('\n', remove_empty: true)
+    end
+
+    def tags_by_branch(branch)
+      return exec "git tag --merged #{branch}"
+    end
+
+    def current_branch
+      return (exec "git symbolic-ref --short HEAD")[0]
+    end
+
+    def current_commit_hash : String
+      cmd = "git rev-parse --verify HEAD --short"
+      return (exec cmd)[0].rjust(7, '0')
+    end
+
+    def get_bumps(latest)
+      begin
+        last_commit = (exec "git show-ref -s #{latest}")[0]
+        # last_commit = (exec "git merge-base #{base_commit} #{current_branch()}")[0]
+        return (exec "git log --pretty=%B #{last_commit}..HEAD")
+      rescue
+        return [] of String
+      end
+    end
+
+    def get_version
+      cb = current_branch
+
+      masterTags = tags_by_branch(cb)
```

Comment:
Given the name, I thought that this method would return all/some of the tags for a given branch. 
It seems to be returning the tags on merge commits, am I right?

I find this names (masterTags and tags_by_branch) a bit confusing